### PR TITLE
XEP-0377: complete rework based on list feedback

### PR DESCRIPTION
**Pull request by SamWhited**
This is a complete rework of the XEP based on list feedback.

Highlights include:

- New payload format
- Ability to include specific stanza-ids to report.
- More guidelines for implementations/UI

### CODE REVIEW

File: xep-0377.xml [hunk 1]
```diff
@@ -127,27 +142,57 @@
...
+    user wishes to report as a child of the &lt;report/&gt; element.
+    The stanza indicated by the provided stanza-id SHOULD be by the same JID
+    being reported and blocked.
   </p>
+  <example caption='Report sent with stanza IDs'><![CDATA[
+<iq from='juliet@example.com/chamber' type='set' id='block1'>
+  <block xmlns='urn:xmpp:blocking'>
+    <item jid='romeo@example.net'>
+      <report xmlns="urn:xmpp:reporting:1" reason="urn:xmpp:reporting:spam">
+        <stanza-id xmlns='urn:xmpp:sid:0' by='romeo@example.net' id='28482-98726-73623'/>
```
Comment (Flowdalic) [inline xep-0377.xml:171]: +1, finally some XEP that does references-by-xep359-stanza-id correctly 🎉

**Comment by horazont** [comment]
Needs Editor Action because a status change is required.

**Comment by SamWhited** [comment]
*sigh* I do that every time. Fixed the status.

**Comment by horazont** [comment]
@SamWhited Thanks! I would’ve done it in the process otherwise :)

**Comment by SamWhited** [comment]
Already merged as 7bcd3330e26804cbc98d1c532fed54c352518907

**Pull request closed without merging**